08_AdvancedTypes/UniversalTablePrinter.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Reflection;
+using System.Text;
+
+public class UniversalTablePrinter
+{
+    private readonly IEnumerable<object> _data;
+    private const int _columnLength = 20;
+
+    public UniversalTablePrinter(IEnumerable<object> data)
+    {
+        _data = data;
+    }
+    public void PrintToConsole()
+    {
+        object firstElement = _data.First();
```

Comment:
if your first object is null your method fails, if you know every item is of the same type, which is needed for the types to match into columns, you can go directly into the type, using the generic example would be `Type type = typeof(T)`